signer/src/stacks/api.rs

```diff
@@ -1146,20 +1219,20 @@ where
         }
         // Tenure blocks are always non-empty, and this invariant is upheld
         // by the type. So no need to worry about the early break.
-        let Some(block) = tenure.blocks().last() else {
+        let Some(header) = tenure.headers().last() else {
             break;
         };
         // We've seen this parent already, so time to stop.
-        if db.stacks_block_exists(block.header.parent_block_id).await? {
+        if db.stacks_block_exists(header.parent_block_id).await? {
             tracing::debug!("parent block known in the database");
             break;
         }
         // There are more blocks to fetch, so let's get them.
-        let tenure_blocks = stacks.get_tenure(block.header.parent_block_id).await?;
-        blocks.push(tenure_blocks);
+        let tenure_blocks = stacks.get_tenure(header.parent_block_id).await?;
+        headers.push(tenure_blocks.into());
```

Comment:
Maybe we should update the fn docs to explicitly mention that the returned headers are in descending order, and also change the doc to not mention a datastore since this method has no knowledge of such.

Reply:
Yeah, make sense. We should also return them in ascending order, it doesn't matter for the current code but maybe it does for some newer code.

As for the datastore mention, this method needs something that implements `DbRead` as an input, so we should probably mention why in the docs (or at least not remove it). Or am I misunderstanding what your asking.

Reply:
Done in https://github.com/stacks-sbtc/sbtc/pull/1604/commits/1c59451abb9014900e3c5f955ef3062f2c3bc129.